### DISCUSSION TITLE
Failed compile mxnet master with cuda9.2 and ubuntu 18.04

### DISCUSSION THREAD

**Discussion by guanlinting** [apache/incubator-mxnet#12241] [Q&A]
## Description
Failed compile mxnet master with cuda9.2 and ubuntu 18.04.

Error messages are as follow:
build/src/operator/nn/deconvolution_gpu.o：在函数‘mxnet::op::CuDNNDeconvolutionOp<float>::SelectAlgo(mxnet::RunContext const&, std::vector<nnvm::TShape, std::allocator<nnvm::TShape> > const&, std::vector<nnvm::TShape, std::allocator<nnvm::TShape> > const&, cudnnDataType_t, cudnnDataType_t)’中：
tmpxft_00005635_00000000-5_deconvolution.compute_70.cudafe1.cpp:(.text._ZN5mxnet2op20CuDNNDeconvolutionOpIfE10SelectAlgoERKNS_10RunContextERKSt6vectorIN4nnvm6TShapeESaIS8_EESC_15cudnnDataType_tSD_[_ZN5mxnet2op20CuDNNDeconvolutionOpIfE10SelectAlgoERKNS_10RunContextERKSt6vectorIN4nnvm6TShapeESaIS8_EESC_15cudnnDataType_tSD_]+0xbe)：对‘mxnet::op::CuDNNAlgoReg<mxnet::op::DeconvolutionParam>::Get()’未定义的引用
tmpxft_00005635_00000000-5_deconvolution.compute_70.cudafe1.cpp:(.text._ZN5mxnet2op20CuDNNDeconvolutionOpIfE10SelectAlgoERKNS_10RunContextERKSt6vectorIN4nnvm6TShapeESaIS8_EESC_15cudnnDataType_tSD_[_ZN5mxnet2op20CuDNNDeconvolutionOpIfE10SelectAlgoERKNS_10RunContextERKSt6vectorIN4nnvm6TShapeESaIS8_EESC_15cudnnDataType_tSD_]+0xd66)：对‘mxnet::op::CuDNNAlgoReg<mxnet::op::DeconvolutionParam>::Get()’未定义的引用
build/src/operator/nn/deconvolution_gpu.o：在函数‘mxnet::op::CuDNNDeconvolutionOp<double>::SelectAlgo(mxnet::RunContext const&, std::vector<nnvm::TShape, std::allocator<nnvm::TShape> > const&, std::vector<nnvm::TShape, std::allocator<nnvm::TShape> > const&, cudnnDataType_t, cudnnDataType_t)’中：
tmpxft_00005635_00000000-5_deconvolution.compute_70.cudafe1.cpp:(.text._ZN5mxnet2op20CuDNNDeconvolutionOpIdE10SelectAlgoERKNS_10RunContextERKSt6vectorIN4nnvm6TShapeESaIS8_EESC_15cudnnDataType_tSD_[_ZN5mxnet2op20CuDNNDeconvolutionOpIdE10SelectAlgoERKNS_10RunContextERKSt6vectorIN4nnvm6TShapeESaIS8_EESC_15cudnnDataType_tSD_]+0xbe)：对‘mxnet::op::CuDNNAlgoReg<mxnet::op::DeconvolutionParam>::Get()’未定义的引用
tmpxft_00005635_00000000-5_deconvolution.compute_70.cudafe1.cpp:(.text._ZN5mxnet2op20CuDNNDeconvolutionOpIdE10SelectAlgoERKNS_10RunContextERKSt6vectorIN4nnvm6TShapeESaIS8_EESC_15cudnnDataType_tSD_[_ZN5mxnet2op20CuDNNDeconvolutionOpIdE10SelectAlgoERKNS_10RunContextERKSt6vectorIN4nnvm6TShapeESaIS8_EESC_15cudnnDataType_tSD_]+0xd66)：对‘mxnet::op::CuDNNAlgoReg<mxnet::op::DeconvolutionParam>::Get()’未定义的引用
build/src/operator/nn/deconvolution_gpu.o：在函数‘mxnet::op::CuDNNDeconvolutionOp<mshadow::half::half_t>::SelectAlgo(mxnet::RunContext const&, std::vector<nnvm::TShape, std::allocator<nnvm::TShape> > const&, std::vector<nnvm::TShape, std::allocator<nnvm::TShape> > const&, cudnnDataType_t, cudnnDataType_t)’中：
tmpxft_00005635_00000000-5_deconvolution.compute_70.cudafe1.cpp:(.text._ZN5mxnet2op20CuDNNDeconvolutionOpIN7mshadow4half6half_tEE10SelectAlgoERKNS_10RunContextERKSt6vectorIN4nnvm6TShapeESaISB_EESF_15cudnnDataType_tSG_[_ZN5mxnet2op20CuDNNDeconvolutionOpIN7mshadow4half6half_tEE10SelectAlgoERKNS_10RunContextERKSt6vectorIN4nnvm6TShapeESaISB_EESF_15cudnnDataType_tSG_]+0xbe)：对‘mxnet::op::CuDNNAlgoReg<mxnet::op::DeconvolutionParam>::Get()’未定义的引用
build/src/operator/nn/deconvolution_gpu.o:tmpxft_00005635_00000000-5_deconvolution.compute_70.cudafe1.cpp:(.text._ZN5mxnet2op20CuDNNDeconvolutionOpIN7mshadow4half6half_tEE10SelectAlgoERKNS_10RunContextERKSt6vectorIN4nnvm6TShapeESaISB_EESF_15cudnnDataType_tSG_[_ZN5mxnet2op20CuDNNDeconvolutionOpIN7mshadow4half6half_tEE10SelectAlgoERKNS_10RunContextERKSt6vectorIN4nnvm6TShapeESaISB_EESF_15cudnnDataType_tSG_]+0xd66): 跟着更多未定义的参考到 mxnet::op::CuDNNAlgoReg<mxnet::op::DeconvolutionParam>::Get()
build/src/operator/nn/convolution_gpu.o：在函数‘mxnet::op::CuDNNConvolutionOp<float>::SelectAlgo(mxnet::RunContext const&, std::vector<nnvm::TShape, std::allocator<nnvm::TShape> > const&, std::vector<nnvm::TShape, std::allocator<nnvm::TShape> > const&, cudnnDataType_t, cudnnDataType_t)’中：
tmpxft_00005904_00000000-5_convolution.compute_70.cudafe1.cpp:(.text._ZN5mxnet2op18CuDNNConvolutionOpIfE10SelectAlgoERKNS_10RunContextERKSt6vectorIN4nnvm6TShapeESaIS8_EESC_15cudnnDataType_tSD_[_ZN5mxnet2op18CuDNNConvolutionOpIfE10SelectAlgoERKNS_10RunContextERKSt6vectorIN4nnvm6TShapeESaIS8_EESC_15cudnnDataType_tSD_]+0x219)：对‘mxnet::op::CuDNNAlgoReg<mxnet::op::ConvolutionParam>::Get()’未定义的引用
tmpxft_00005904_00000000-5_convolution.compute_70.cudafe1.cpp:(.text._ZN5mxnet2op18CuDNNConvolutionOpIfE10SelectAlgoERKNS_10RunContextERKSt6vectorIN4nnvm6TShapeESaIS8_EESC_15cudnnDataType_tSD_[_ZN5mxnet2op18CuDNNConvolutionOpIfE10SelectAlgoERKNS_10RunContextERKSt6vectorIN4nnvm6TShapeESaIS8_EESC_15cudnnDataType_tSD_]+0x170c)：对‘mxnet::op::CuDNNAlgoReg<mxnet::op::ConvolutionParam>::Get()’未定义的引用
build/src/operator/nn/convolution_gpu.o：在函数‘mxnet::op::CuDNNConvolutionOp<double>::SelectAlgo(mxnet::RunContext const&, std::vector<nnvm::TShape, std::allocator<nnvm::TShape> > const&, std::vector<nnvm::TShape, std::allocator<nnvm::TShape> > const&, cudnnDataType_t, cudnnDataType_t)’中：
tmpxft_00005904_00000000-5_convolution.compute_70.cudafe1.cpp:(.text._ZN5mxnet2op18CuDNNConvolutionOpIdE10SelectAlgoERKNS_10RunContextERKSt6vectorIN4nnvm6TShapeESaIS8_EESC_15cudnnDataType_tSD_[_ZN5mxnet2op18CuDNNConvolutionOpIdE10SelectAlgoERKNS_10RunContextERKSt6vectorIN4nnvm6TShapeESaIS8_EESC_15cudnnDataType_tSD_]+0x219)：对‘mxnet::op::CuDNNAlgoReg<mxnet::op::ConvolutionParam>::Get()’未定义的引用
tmpxft_00005904_00000000-5_convolution.compute_70.cudafe1.cpp:(.text._ZN5mxnet2op18CuDNNConvolutionOpIdE10SelectAlgoERKNS_10RunContextERKSt6vectorIN4nnvm6TShapeESaIS8_EESC_15cudnnDataType_tSD_[_ZN5mxnet2op18CuDNNConvolutionOpIdE10SelectAlgoERKNS_10RunContextERKSt6vectorIN4nnvm6TShapeESaIS8_EESC_15cudnnDataType_tSD_]+0x170c)：对‘mxnet::op::CuDNNAlgoReg<mxnet::op::ConvolutionParam>::Get()’未定义的引用
build/src/operator/nn/convolution_gpu.o：在函数‘mxnet::op::CuDNNConvolutionOp<mshadow::half::half_t>::SelectAlgo(mxnet::RunContext const&, std::vector<nnvm::TShape, std::allocator<nnvm::TShape> > const&, std::vector<nnvm::TShape, std::allocator<nnvm::TShape> > const&, cudnnDataType_t, cudnnDataType_t)’中：
tmpxft_00005904_00000000-5_convolution.compute_70.cudafe1.cpp:(.text._ZN5mxnet2op18CuDNNConvolutionOpIN7mshadow4half6half_tEE10SelectAlgoERKNS_10RunContextERKSt6vectorIN4nnvm6TShapeESaISB_EESF_15cudnnDataType_tSG_[_ZN5mxnet2op18CuDNNConvolutionOpIN7mshadow4half6half_tEE10SelectAlgoERKNS_10RunContextERKSt6vectorIN4nnvm6TShapeESaISB_EESF_15cudnnDataType_tSG_]+0x219)：对‘mxnet::op::CuDNNAlgoReg<mxnet::op::ConvolutionParam>::Get()’未定义的引用
build/src/operator/nn/convolution_gpu.o:tmpxft_00005904_00000000-5_convolution.compute_70.cudafe1.cpp:(.text._ZN5mxnet2op18CuDNNConvolutionOpIN7mshadow4half6half_tEE10SelectAlgoERKNS_10RunContextERKSt6vectorIN4nnvm6TShapeESaISB_EESF_15cudnnDataType_tSG_[_ZN5mxnet2op18CuDNNConvolutionOpIN7mshadow4half6half_tEE10SelectAlgoERKNS_10RunContextERKSt6vectorIN4nnvm6TShapeESaISB_EESF_15cudnnDataType_tSG_]+0x170c): 跟着更多未定义的参考到 mxnet::op::CuDNNAlgoReg<mxnet::op::ConvolutionParam>::Get()
collect2: error: ld returned 1 exit status
Makefile:515: recipe for target 'bin/im2rec' failed
make: *** [bin/im2rec] Error 1

**Comment by haojin2** [comment]
Can you also provide more details on your environment?
Detailed instructions provided in the template of issues, copying here for your convenience:
```
What to do:
1. Download the diagnosis script from https://raw.githubusercontent.com/apache/incubator-mxnet/master/tools/diagnose.py
2. Run the script using `python diagnose.py` and paste its output here.
```

**Comment by vdantu** [comment]
@guanlinting : Could you please provide further information? 

@mxnet-label-bot : [Ubuntu, Build, CUDA, Question]

**Comment by guanlinting** [comment]
ok haojin2, here is the output of the script:
----------Python Info----------
('Version      :', '2.7.15rc1')
('Compiler     :', 'GCC 7.3.0')
('Build        :', ('default', 'Apr 15 2018 21:51:34'))
('Arch         :', ('64bit', ''))
------------Pip Info-----------
('Version      :', '9.0.1')
('Directory    :', '/usr/lib/python2.7/dist-packages/pip')
----------MXNet Info-----------
/home/bonobo/.local/lib/python2.7/site-packages/scipy-1.1.0-py2.7-linux-x86_64.egg/scipy/sparse/lil.py:19: RuntimeWarning: numpy.dtype size changed, may indicate binary incompatibility. Expected 96, got 88
  from . import _csparsetools
/home/bonobo/.local/lib/python2.7/site-packages/scipy-1.1.0-py2.7-linux-x86_64.egg/scipy/sparse/csgraph/__init__.py:165: RuntimeWarning: numpy.dtype size changed, may indicate binary incompatibility. Expected 96, got 88
  from ._shortest_path import shortest_path, floyd_warshall, dijkstra,\
/home/bonobo/.local/lib/python2.7/site-packages/scipy-1.1.0-py2.7-linux-x86_64.egg/scipy/sparse/csgraph/_validation.py:5: RuntimeWarning: numpy.dtype size changed, may indicate binary incompatibility. Expected 96, got 88
  from ._tools import csgraph_to_dense, csgraph_from_dense,\
/home/bonobo/.local/lib/python2.7/site-packages/scipy-1.1.0-py2.7-linux-x86_64.egg/scipy/sparse/csgraph/__init__.py:167: RuntimeWarning: numpy.dtype size changed, may indicate binary incompatibility. Expected 96, got 88
  from ._traversal import breadth_first_order, depth_first_order, \
/home/bonobo/.local/lib/python2.7/site-packages/scipy-1.1.0-py2.7-linux-x86_64.egg/scipy/sparse/csgraph/__init__.py:169: RuntimeWarning: numpy.dtype size changed, may indicate binary incompatibility. Expected 96, got 88
  from ._min_spanning_tree import minimum_spanning_tree
/home/bonobo/.local/lib/python2.7/site-packages/scipy-1.1.0-py2.7-linux-x86_64.egg/scipy/sparse/csgraph/__init__.py:170: RuntimeWarning: numpy.dtype size changed, may indicate binary incompatibility. Expected 96, got 88
  from ._reordering import reverse_cuthill_mckee, maximum_bipartite_matching, \
/home/bonobo/.local/lib/python2.7/site-packages/scipy-1.1.0-py2.7-linux-x86_64.egg/scipy/special/__init__.py:640: RuntimeWarning: numpy.dtype size changed, may indicate binary incompatibility. Expected 96, got 88
  from ._ufuncs import *
/home/bonobo/.local/lib/python2.7/site-packages/scipy-1.1.0-py2.7-linux-x86_64.egg/scipy/linalg/basic.py:17: RuntimeWarning: numpy.dtype size changed, may indicate binary incompatibility. Expected 96, got 88
  from ._solve_toeplitz import levinson
/home/bonobo/.local/lib/python2.7/site-packages/scipy-1.1.0-py2.7-linux-x86_64.egg/scipy/linalg/__init__.py:207: RuntimeWarning: numpy.dtype size changed, may indicate binary incompatibility. Expected 96, got 88
  from ._decomp_update import *
/home/bonobo/.local/lib/python2.7/site-packages/scipy-1.1.0-py2.7-linux-x86_64.egg/scipy/special/_ellip_harm.py:7: RuntimeWarning: numpy.dtype size changed, may indicate binary incompatibility. Expected 96, got 88
  from ._ellip_harm_2 import _ellipsoid, _ellipsoid_norm
/home/bonobo/.local/lib/python2.7/site-packages/scipy-1.1.0-py2.7-linux-x86_64.egg/scipy/interpolate/_bsplines.py:10: RuntimeWarning: numpy.dtype size changed, may indicate binary incompatibility. Expected 96, got 88
  from . import _bspl
/home/bonobo/.local/lib/python2.7/site-packages/scipy-1.1.0-py2.7-linux-x86_64.egg/scipy/spatial/__init__.py:95: RuntimeWarning: numpy.dtype size changed, may indicate binary incompatibility. Expected 96, got 88
  from .ckdtree import *
/home/bonobo/.local/lib/python2.7/site-packages/scipy-1.1.0-py2.7-linux-x86_64.egg/scipy/spatial/__init__.py:96: RuntimeWarning: numpy.dtype size changed, may indicate binary incompatibility. Expected 96, got 88
  from .qhull import *
/home/bonobo/.local/lib/python2.7/site-packages/scipy-1.1.0-py2.7-linux-x86_64.egg/scipy/spatial/_spherical_voronoi.py:18: RuntimeWarning: numpy.dtype size changed, may indicate binary incompatibility. Expected 96, got 88
  from . import _voronoi
/home/bonobo/.local/lib/python2.7/site-packages/scipy-1.1.0-py2.7-linux-x86_64.egg/scipy/spatial/distance.py:122: RuntimeWarning: numpy.dtype size changed, may indicate binary incompatibility. Expected 96, got 88
  from . import _hausdorff
/home/bonobo/.local/lib/python2.7/site-packages/scipy-1.1.0-py2.7-linux-x86_64.egg/scipy/optimize/_trlib/__init__.py:1: RuntimeWarning: numpy.dtype size changed, may indicate binary incompatibility. Expected 96, got 88
  from ._trlib import TRLIBQuadraticSubproblem
/home/bonobo/.local/lib/python2.7/site-packages/scipy-1.1.0-py2.7-linux-x86_64.egg/scipy/optimize/_numdiff.py:10: RuntimeWarning: numpy.dtype size changed, may indicate binary incompatibility. Expected 96, got 88
  from ._group_columns import group_dense, group_sparse
/home/bonobo/.local/lib/python2.7/site-packages/scipy-1.1.0-py2.7-linux-x86_64.egg/scipy/stats/_continuous_distns.py:18: RuntimeWarning: numpy.dtype size changed, may indicate binary incompatibility. Expected 96, got 88
  from . import _stats
('Version      :', '1.3.0')
('Directory    :', '/home/bonobo/incubator-mxnet/python/mxnet')
Hashtag not found. Not installed from pre-built package.
----------System Info----------
('Platform     :', 'Linux-4.15.0-32-generic-x86_64-with-Ubuntu-18.04-bionic')
('system       :', 'Linux')
('node         :', 'bonobo')
('release      :', '4.15.0-32-generic')
('version      :', '#35-Ubuntu SMP Fri Aug 10 17:58:07 UTC 2018')
----------Hardware Info----------
('machine      :', 'x86_64')
('processor    :', 'x86_64')
架构：           x86_64
CPU 运行模式：   32-bit, 64-bit
字节序：         Little Endian
CPU:             6
在线 CPU 列表：  0-5
每个核的线程数： 1
每个座的核数：   6
座：             1
NUMA 节点：      1
厂商 ID：        GenuineIntel
CPU 系列：       6
型号：           158
型号名称：       Intel(R) Core(TM) i5-8400 CPU @ 2.80GHz
步进：           10
CPU MHz：        800.049
CPU 最大 MHz：   4000.0000
CPU 最小 MHz：   800.0000
BogoMIPS：       5616.00
虚拟化：         VT-x
L1d 缓存：       32K
L1i 缓存：       32K
L2 缓存：        256K
L3 缓存：        9216K
NUMA 节点0 CPU： 0-5
标记：           fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf tsc_known_freq pni pclmulqdq dtes64 monitor ds_cpl vmx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb invpcid_single pti ibrs ibpb stibp tpr_shadow vnmi flexpriority ept vpid fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid mpx rdseed adx smap clflushopt intel_pt xsaveopt xsavec xgetbv1 xsaves dtherm ida arat pln pts hwp hwp_notify hwp_act_window hwp_epp
----------Network Test----------
Setting timeout: 10
Timing for MXNet: https://github.com/apache/incubator-mxnet, DNS: 0.0085 sec, LOAD: 1.1394 sec.
Timing for PYPI: https://pypi.python.org/pypi/pip, DNS: 0.0076 sec, LOAD: 8.3496 sec.
Timing for FashionMNIST: https://apache-mxnet.s3-accelerate.dualstack.amazonaws.com/gluon/dataset/fashion-mnist/train-labels-idx1-ubyte.gz, DNS: 5.0123 sec, LOAD: 2.5292 sec.
Timing for Conda: https://repo.continuum.io/pkgs/free/, DNS: 0.0073 sec, LOAD: 1.6735 sec.
Timing for Gluon Tutorial(en): http://gluon.mxnet.io, DNS: 0.5671 sec, LOAD: 1.5370 sec.
Timing for Gluon Tutorial(cn): https://zh.gluon.ai, DNS: 5.5657 sec, LOAD: 3.7741 sec.
